lib/elixir/lib/map.ex

```diff
@@ -211,13 +220,16 @@ defmodule Map do
   """
   @spec take(map, [key]) :: map
   def take(map, keys) do
-    Enum.reduce(keys, [], fn key, acc ->
-      case fetch(map, key) do
-        {:ok, value} -> [{key, value} | acc]
-        :error -> acc
-      end
-    end)
-    |> :maps.from_list
+    do_take(keys, map, [])
+  end
```

Comment:
I'm gonna go ahead with these changes and fix the specs in the process. I'm noticing that there aren't any tests in `map_test.exs` for these functions. Should I add a doc test example using an enumerable other than a list? Or should I make further tests in `map_test.exs`